@tyoungsc The failed test log is the following:
```
FAIL: SYCL :: Basic/fpga_tests/fpga_pipes.cpp (132 of 809)
[2022-03-16T05:56:35.057Z] ******************** TEST 'SYCL :: Basic/fpga_tests/fpga_pipes.cpp' FAILED ********************
[2022-03-16T05:56:35.057Z] Script:
[2022-03-16T05:56:35.057Z] --
[2022-03-16T05:56:35.057Z] : 'RUN: at line 1';    /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm.obj/bin/clang++      -fsycl /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp -o /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/build/SYCL/Basic/fpga_tests/Output/fpga_pipes.cpp.tmp.out
[2022-03-16T05:56:35.057Z] : 'RUN: at line 3';   true /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/build/SYCL/Basic/fpga_tests/Output/fpga_pipes.cpp.tmp.out
[2022-03-16T05:56:35.057Z] : 'RUN: at line 4';    env SYCL_DEVICE_FILTER=ext_oneapi_level_zero:gpu,host  /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/build/SYCL/Basic/fpga_tests/Output/fpga_pipes.cpp.tmp.out
[2022-03-16T05:56:35.057Z] : 'RUN: at line 5';   true /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/build/SYCL/Basic/fpga_tests/Output/fpga_pipes.cpp.tmp.out
[2022-03-16T05:56:35.057Z] --
[2022-03-16T05:56:35.057Z] Exit Code: 1
[2022-03-16T05:56:35.057Z] 
[2022-03-16T05:56:35.057Z] Command Output (stdout):
[2022-03-16T05:56:35.057Z] --
[2022-03-16T05:56:35.057Z] $ ":" "RUN: at line 1"
[2022-03-16T05:56:35.057Z] note: command had no output on stdout or stderr
[2022-03-16T05:56:35.057Z] $ "/runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm.obj/bin/clang++" "-fsycl" "/runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp" "-o" "/runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/build/SYCL/Basic/fpga_tests/Output/fpga_pipes.cpp.tmp.out"
[2022-03-16T05:56:35.057Z] # command stderr:
[2022-03-16T05:56:35.057Z] /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp:36:30: error: 'PipeMulNb' is not a class, namespace, or enumeration
[2022-03-16T05:56:35.057Z] static_assert(std::is_same_v<PipeMulNb::value_type, int>);
[2022-03-16T05:56:35.057Z]                              ^
[2022-03-16T05:56:35.057Z] /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp:35:1: note: 'PipeMulNb' declared here
[2022-03-16T05:56:35.057Z] using PipeMulNb = cl::sycl::ext::intel::pipe<class templ_nb_pipe<N>, int>;
[2022-03-16T05:56:35.057Z] ^
[2022-03-16T05:56:35.057Z] /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp:37:15: error: 'PipeMulNb' is not a class, namespace, or enumeration
[2022-03-16T05:56:35.057Z] static_assert(PipeMulNb::min_capacity == 0);
[2022-03-16T05:56:35.057Z]               ^
[2022-03-16T05:56:35.057Z] /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp:35:1: note: 'PipeMulNb' declared here
[2022-03-16T05:56:35.057Z] using PipeMulNb = cl::sycl::ext::intel::pipe<class templ_nb_pipe<N>, int>;
[2022-03-16T05:56:35.057Z] ^
[2022-03-16T05:56:35.057Z] /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp:53:30: error: 'PipeMulBl' is not a class, namespace, or enumeration
[2022-03-16T05:56:35.057Z] static_assert(std::is_same_v<PipeMulBl::value_type, int>);
[2022-03-16T05:56:35.057Z]                              ^
[2022-03-16T05:56:35.057Z] /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp:52:1: note: 'PipeMulBl' declared here
[2022-03-16T05:56:35.057Z] using PipeMulBl = cl::sycl::ext::intel::pipe<class templ_bl_pipe<N>, int>;
[2022-03-16T05:56:35.057Z] ^
[2022-03-16T05:56:35.057Z] /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp:54:15: error: 'PipeMulBl' is not a class, namespace, or enumeration
[2022-03-16T05:56:35.057Z] static_assert(PipeMulBl::min_capacity == 0);
[2022-03-16T05:56:35.057Z]               ^
[2022-03-16T05:56:35.058Z] /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp:52:1: note: 'PipeMulBl' declared here
[2022-03-16T05:56:35.058Z] using PipeMulBl = cl::sycl::ext::intel::pipe<class templ_bl_pipe<N>, int>;
[2022-03-16T05:56:35.058Z] ^
[2022-03-16T05:56:35.058Z] /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp:66:32: error: template argument for template type parameter must be a type; did you forget 'typename'?
[2022-03-16T05:56:35.058Z]   static_assert(std::is_same_v<Pipe::value_type, int>);
[2022-03-16T05:56:35.058Z]                                ^
[2022-03-16T05:56:35.058Z]                                typename 
[2022-03-16T05:56:35.058Z] /usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/type_traits:2995:20: note: template parameter is declared here
[2022-03-16T05:56:35.058Z] template <typename _Tp, typename _Up>
[2022-03-16T05:56:35.058Z]                    ^
[2022-03-16T05:56:35.058Z] /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp:157:32: error: template argument for template type parameter must be a type; did you forget 'typename'?
[2022-03-16T05:56:35.058Z]   static_assert(std::is_same_v<AnotherNbPipe::value_type, int>);
[2022-03-16T05:56:35.058Z]                                ^
[2022-03-16T05:56:35.058Z]                                typename 
[2022-03-16T05:56:35.058Z] /usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/type_traits:2995:20: note: template parameter is declared here
[2022-03-16T05:56:35.058Z] template <typename _Tp, typename _Up>
[2022-03-16T05:56:35.058Z]                    ^
[2022-03-16T05:56:35.058Z] /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp:201:32: error: template argument for template type parameter must be a type; did you forget 'typename'?
[2022-03-16T05:56:35.058Z]   static_assert(std::is_same_v<Pipe::value_type, int>);
[2022-03-16T05:56:35.058Z]                                ^
[2022-03-16T05:56:35.058Z]                                typename 
[2022-03-16T05:56:35.058Z] /usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/type_traits:2995:20: note: template parameter is declared here
[2022-03-16T05:56:35.058Z] template <typename _Tp, typename _Up>
[2022-03-16T05:56:35.058Z]                    ^
[2022-03-16T05:56:35.058Z] /runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Basic/fpga_tests/fpga_pipes.cpp:271:32: error: template argument for template type parameter must be a type; did you forget 'typename'?
[2022-03-16T05:56:35.058Z]   static_assert(std::is_same_v<AnotherBlPipe::value_type, int>);
[2022-03-16T05:56:35.058Z]                                ^
[2022-03-16T05:56:35.058Z]                                typename 
[2022-03-16T05:56:35.058Z] /usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/type_traits:2995:20: note: template parameter is declared here
[2022-03-16T05:56:35.058Z] template <typename _Tp, typename _Up>
[2022-03-16T05:56:35.058Z]                    ^
[2022-03-16T05:56:35.058Z] 8 errors generated.
[2022-03-16T05:56:35.058Z] 
[2022-03-16T05:56:35.058Z] error: command failed with exit status: 1
[2022-03-16T05:56:35.058Z] 
[2022-03-16T05:56:35.058Z] --
[2022-03-16T05:56:35.058Z] 
[2022-03-16T05:56:35.058Z] ********************
```

--- a/SYCL/Basic/fpga_tests/fpga_pipes.cpp
+++ b/SYCL/Basic/fpga_tests/fpga_pipes.cpp
@@ -33,8 +33,8 @@ template <int N> class templ_nb_pipe;
 // For non-blocking multiple pipes
 template <int N>
 using PipeMulNb = cl::sycl::ext::intel::pipe<class templ_nb_pipe<N>, int>;
-static_assert(std::is_same_v<PipeMulNb::value_type, int>);
-static_assert(PipeMulNb::min_capacity == 0);
+static_assert(std::is_same_v<typename PipeMulNb<0>::value_type, int>);
+static_assert(PipeMulNb<0>::min_capacity == 0);
 
 // For simple blocking pipes with explicit type
 class some_bl_pipe;
@@ -50,8 +50,8 @@ template <int N> class templ_bl_pipe;
 // For blocking multiple pipes
 template <int N>
 using PipeMulBl = cl::sycl::ext::intel::pipe<class templ_bl_pipe<N>, int>;
-static_assert(std::is_same_v<PipeMulBl::value_type, int>);
-static_assert(PipeMulBl::min_capacity == 0);
+static_assert(std::is_same_v<typename PipeMulBl<0>::value_type, int>);
+static_assert(PipeMulBl<0>::min_capacity == 0);
 
 // Kernel names
 template <int TestNumber, int KernelNumber = 0> class writer;
@@ -63,7 +63,7 @@ int test_simple_nb_pipe(cl::sycl::queue Queue) {
   int data[] = {0};
 
   using Pipe = cl::sycl::ext::intel::pipe<PipeName, int>;
-  static_assert(std::is_same_v<Pipe::value_type, int>);
+  static_assert(std::is_same_v<typename Pipe::value_type, int>);
   static_assert(Pipe::min_capacity == 0);
 
   cl::sycl::buffer<int, 1> readBuf(data, 1);
@@ -154,7 +154,7 @@ template <int TestNumber> int test_multiple_nb_pipe(cl::sycl::queue Queue) {
 template <int TestNumber> int test_array_th_nb_pipe(cl::sycl::queue Queue) {
   int data[N] = {0};
   using AnotherNbPipe = cl::sycl::ext::intel::pipe<class another_nb_pipe, int>;
-  static_assert(std::is_same_v<AnotherNbPipe::value_type, int>);
+  static_assert(std::is_same_v<typename AnotherNbPipe::value_type, int>);
   static_assert(AnotherNbPipe::min_capacity == 0);
 
   Queue.submit([&](cl::sycl::handler &cgh) {
@@ -198,7 +198,7 @@ int test_simple_bl_pipe(cl::sycl::queue Queue) {
   int data[] = {0};
 
   using Pipe = cl::sycl::ext::intel::pipe<PipeName, int>;
-  static_assert(std::is_same_v<Pipe::value_type, int>);
+  static_assert(std::is_same_v<typename Pipe::value_type, int>);
   static_assert(Pipe::min_capacity == 0);
 
   cl::sycl::buffer<int, 1> readBuf(data, 1);
@@ -268,7 +268,7 @@ template <int TestNumber> int test_multiple_bl_pipe(cl::sycl::queue Queue) {
 template <int TestNumber> int test_array_th_bl_pipe(cl::sycl::queue Queue) {
   int data[N] = {0};
   using AnotherBlPipe = cl::sycl::ext::intel::pipe<class another_bl_pipe, int>;
-  static_assert(std::is_same_v<AnotherBlPipe::value_type, int>);
+  static_assert(std::is_same_v<typename AnotherBlPipe::value_type, int>);
   static_assert(AnotherBlPipe::min_capacity == 0);
 
   Queue.submit([&](cl::sycl::handler &cgh) {

